### PR TITLE
Ping when opening SQL connection

### PR DESCRIPTION
**Pull request by sethvargo**
Also reduce shadowing and increase total allowed timeout to 30s.

I've noticed an increase in test flakes, likely due to the increase in test coverage and additional overhead of creating more database instances.

**Release Note**

```release-note
Ping when opening SQL connection
```

### CODE REVIEW

File: pkg/database/database.go [hunk 1]
```diff
@@ -149,33 +149,39 @@ func (db *Database) OpenWithCacher(ctx context.Context, cacher cache.Cacher) err
 	// opencenusus stats.
 	var rawSQL *sql.DB
 	if err := withRetries(ctx, func(ctx context.Context) error {
-		var err error
-		rawSQL, err = sql.Open("ocsql", c.ConnectionString())
+		s, err := sql.Open("ocsql", c.ConnectionString())
 		if err != nil {
+			return fmt.Errorf("failed to open sql connection: %w", err)
```
Comment (sethvargo) [inline pkg/database/database.go:154]: This is intentionally not retryable. "Open" doesn't test the connection, it just parses the string and creates a struct.

File: pkg/database/database.go [hunk 2]
```diff
@@ -149,33 +149,39 @@ func (db *Database) OpenWithCacher(ctx context.Context, cacher cache.Cacher) err
...
+		}
+
+		if err := s.Ping(); err != nil {
 			return retry.RetryableError(err)
 		}
-		db.statsCloser = ocsql.RecordStats(rawSQL, 5*time.Second)
```
Comment (sethvargo) [inline pkg/database/database.go:157]: Moved this out of the loop.

File: pkg/database/database.go [hunk 3]
```diff
@@ -149,33 +149,39 @@ func (db *Database) OpenWithCacher(ctx context.Context, cacher cache.Cacher) err
...
 	var rawSQL *sql.DB
 	if err := withRetries(ctx, func(ctx context.Context) error {
-		var err error
-		rawSQL, err = sql.Open("ocsql", c.ConnectionString())
+		s, err := sql.Open("ocsql", c.ConnectionString())
 		if err != nil {
+			return fmt.Errorf("failed to open sql connection: %w", err)
+		}
+
+		if err := s.Ping(); err != nil {
```
Comment (mikehelmick) [inline pkg/database/database.go:157]: does `s` need to be closed if ping fails?

Comment (sethvargo) [reply pkg/database/database.go]: Hmm yea, good point. I wonder if that's why we're exhausting... I refactored this a bit.

**Review by mikehelmick — APPROVED** [review]

**Pull request merged**